### PR TITLE
Add mixed16 tests to collective functions

### PR DESCRIPTION
**Pull request by belldandyxtq**
This commit add mixed16 tests to all the collective functions

Thank you for creating a pull request!

Please double-check the following.

- Read [our contribution guide](https://docs.chainer.org/en/stable/contribution.html).
  - Does your code conform to our coding guidelines?
  - Did you write sufficient test code?
  - Did you write sufficient documentation?
- Also, take a look at [our compatibility policy](https://docs.chainer.org/en/stable/compatibility.html).

### CODE REVIEW

**Comment by belldandyxtq** [comment]
pfnci, test this please

File: tests/chainermn_tests/functions_tests/test_collective_communication.py [hunk 1]
```diff
@@ -53,9 +63,11 @@ def check_all_gather(xs, communicator):
 @pytest.mark.parametrize('param', params)
 def test_all_gather_cpu(param):
     communicator = get_communicator(False)
+    io_dtype = get_io_dtype(param.dtype)
```
Comment (shu65) [inline tests/chainermn_tests/functions_tests/test_collective_communication.py]: How about using `chainer.get_dtype()`? 
ref: https://docs.chainer.org/en/stable/reference/generated/chainer.get_dtype.html

**Comment by belldandyxtq** [comment]
pfnci, test this please

**Comment by shu65** [comment]
I'm performing unittests.

**Comment by shu65** [comment]
pfnci, test this please

**Pull request merged**